std/functional.d

```diff
@@ -992,10 +992,11 @@ template memoize(alias fun)
         alias Args = Parameters!fun;
         import std.typecons : Tuple;
 
-        static ReturnType!fun[Tuple!Args] memo;
-        auto t = Tuple!Args(args);
+        static ReturnType!fun[ulong] memo;
+        auto t = Tuple!Args(args).hashOf;
```

Comment:
Fair enough - so should the result be recomputed and overwritten if `args1 != args2` (like for the limited memoize variant)?